test/tests/bootstrap/test_api20_os_bootstrap_parallel_proxy.py

```diff
@@ -0,0 +1,260 @@
+'''
+Copyright 2017 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+Author(s):
+George Paulos
+
+This script tests minimum payload base case of the RackHD API 2.0 OS bootstrap workflows using proxy method.
+This routine runs OS bootstrap jobs simultaneously on multiple nodes.
+For 11 tests to run, 11 nodes are required in the stack. If there are less than that, tests will be skipped.
+This test takes 15-20 minutes to run.
+
+OS bootstrap tests require special RackHD configuration and mirror repositories for the OS images.
+OS images are loaded via RackHD "httpProxies" settings in config.json.
+
+Example:
+      "httpProxies": [
+        {
+          "localPath": "/ESXi/5.5",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/esxi/5.5/esxi"
+        },
+        {
+          "localPath": "/ESXi/6.0",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/6.0/esxi6"
+        },
+        {
+          "localPath": "/CentOS/6.5",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/centos/6.5/os/x86_64"
+        },
+        {
+          "localPath": "/CentOS/7.0",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/centos/7/os/x86_64"
+        },
+        {
+          "localPath": "/RHEL/7.0",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/rhel/7.0/os/x86_64"
+        },
+        {
+          "localPath": "/coreos",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/mirrors/coreos/17"
+        },
+        {
+          "localPath": "/SLES/12",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/mirrors/sles/12"
+        },
+        {
+          "localPath": "/Ubuntu/14",
+          "remotePath": "/",
+          "server": "http://os-mirror-server/mirrors/ubuntu/boot/14.04.4"
+        }
+      ],
+
+For each OS type, the "localpath" string must conform to the convention above.
+The "server' URL points to the location of the OS executables in an image repository.
+
+'''
+
+import fit_path  # NOQA: unused import
+from nose.plugins.attrib import attr
+import fit_common
+import flogging
+import time
+log = flogging.get_loggers()
+
+# This gets the list of nodes
+NODECATALOG = fit_common.node_select()
+
+# dict containing bootstrap workflow IDs and states
+NODE_STATUS = {}
+
+# global timer
+START_TIME = time.time()
+
+# download RackHD config from host
+rackhdconfig = fit_common.rackhdapi('/api/2.0/config')['json']
```

Comment:
Should probably check if rackhdconfig came back with valid data, ie. status = 200.  relying on it further below in proxy_select.